homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -55,6 +56,32 @@ async def async_setup_entry(
     return True
 
 
+def get_platforms(config_entry):
+    """Return the platforms beloging to a config_entry."""
+    model = config_entry.data[CONF_MODEL]
+    flow_type = config_entry.data[CONF_FLOW_TYPE]
+
+    platforms = []
+
+    if flow_type == CONF_GATEWAY:
+        platforms = GATEWAY_PLATFORMS
+    elif flow_type == CONF_DEVICE:
+        if model in MODELS_SWITCH:
+            platforms = SWITCH_PLATFORMS
+        elif model in MODELS_FAN:
+            platforms = FAN_PLATFORMS
+        elif model in MODELS_LIGHT:
+            platforms = LIGHT_PLATFORMS
+        for vacuum_model in MODELS_VACUUM:
+            if model.startswith(vacuum_model):
+                platforms = VACUUM_PLATFORMS
+        for air_monitor_model in MODELS_AIR_MONITOR:
+            if model.startswith(air_monitor_model):
+                platforms = AIR_MONITOR_PLATFORMS
+
+    return platforms
```

Comment:
I would return from each `if` block instead of nesting here. Then just return `None` at the end in case none of the blocks are met.

```suggestion

    if flow_type == CONF_GATEWAY:
        return GATEWAY_PLATFORMS
    if flow_type == CONF_DEVICE:
        if model in MODELS_SWITCH:
            return SWITCH_PLATFORMS
        if model in MODELS_FAN:
            return FAN_PLATFORMS
        if model in MODELS_LIGHT:
            return LIGHT_PLATFORMS
        for vacuum_model in MODELS_VACUUM:
            if model.startswith(vacuum_model):
                return VACUUM_PLATFORMS
        for air_monitor_model in MODELS_AIR_MONITOR:
            if model.startswith(air_monitor_model):
                return AIR_MONITOR_PLATFORMS
                
    return None
```

Reply:
Thanks for the sugestion